src/plugins/console/public/application/containers/editor/monaco/components/context_menu/context_menu.tsx

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiIcon,
+  EuiContextMenuPanel,
+  EuiContextMenuItem,
+  EuiPopover,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import { NotificationsSetup } from '@kbn/core/public';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { LanguageSelectorModal } from './language_selector_modal';
+import { convertRequestToLanguage } from '../../../../../../services';
+import type { EditorRequest } from '../../types';
+
+import { useServicesContext } from '../../../../../contexts';
+import { StorageKeys } from '../../../../../../services';
+import { DEFAULT_LANGUAGE, AVAILABLE_LANGUAGES } from '../../../../../../../common/constants';
+
+interface Props {
+  getRequests: () => Promise<EditorRequest[]>;
+  getDocumentation: () => Promise<string | null>;
+  autoIndent: (ev: React.MouseEvent) => void;
+  notifications: NotificationsSetup;
+}
+
+const DELAY_FOR_HIDING_SPINNER = 500;
+
+const getLanguageLabelByValue = (value: string) => {
+  return AVAILABLE_LANGUAGES.find((lang) => lang.value === value)?.label || DEFAULT_LANGUAGE;
+};
+
+export const ContextMenu = ({
+  getRequests,
+  getDocumentation,
+  autoIndent,
+  notifications,
+}: Props) => {
+  // Get default language from local storage
+  const {
+    services: { storage, esHostService },
+  } = useServicesContext();
+  const defaultLanguage = storage.get(StorageKeys.DEFAULT_LANGUAGE, DEFAULT_LANGUAGE);
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [isRequestConverterLoading, setRequestConverterLoading] = useState(false);
+  const [isLanguageSelectorVisible, setLanguageSelectorVisibility] = useState(false);
+  const [currentLanguage, setCurrentLanguage] = useState(defaultLanguage);
+
+  const copyText = async (text: string) => {
+    if (window.navigator?.clipboard) {
+      await window.navigator.clipboard.writeText(text);
+      return;
+    }
+    throw new Error('Could not copy to clipboard!');
+  };
+
+  // This function will convert all the selected requests to the language by
+  // calling convertRequestToLanguage for each request and then copy the data
```

Comment:
nit: I think this comment needs to updated, as I understand the change in the implementation was made to call the converter for all requests at once instead of for each.

Reply:
Good catch! Addressed with https://github.com/elastic/kibana/pull/189976/commits/009776f474b69ec425d448de1a50c293c3281ed5